app/services/topic_status_updater.rb

```diff
@@ -46,8 +46,9 @@ def change(status, opts = {})
       UserProfile.remove_featured_topic_from_all_profiles(topic)
     end
 
-    if status.visible?
+    if status.visible? && result
```

Comment:
Another bug fix here. We need to make sure that the status is actually being changed before updating the count. Otherwise, the count may be changed incorrectly if we attempt to update the status of a topic twice.